pyomo/core/base/sets.py

```diff
@@ -110,6 +110,72 @@ def wrapper_function ( *args, **kwargs ):
     return wrapper_function
 
 
+class _robust_sort_keys(object):
```

Comment:
Why is this a class?  Why not provide a function, which is a simpler code design?